src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -97,19 +96,36 @@ public async Task SendAsync_SlowServerAndCancel_ThrowsTaskCanceledException()
             }
         }
 
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/20675")]
         [OuterLoop]
         [Fact]
-        [OuterLoop]
-        public void SendAsync_SlowServerRespondsAfterDefaultReceiveTimeout_ThrowsHttpRequestException()
+        public async void SendAsync_SlowServerRespondsAfterDefaultReceiveTimeout_ThrowsHttpRequestException()
         {
             var handler = new WinHttpHandler();
             using (var client = new HttpClient(handler))
             {
-                Task<HttpResponseMessage> t = client.GetAsync(SlowServer);
+                var triggerResponseWrite = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+                var triggerRequestWait = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
 
-                AggregateException ag = Assert.Throws<AggregateException>(() => t.Wait());
-                Assert.IsType<HttpRequestException>(ag.InnerException);
+                await LoopbackServer.CreateServerAsync(async (server, url) =>
+                {
+                    Task serverTask = server.AcceptConnectionAsync(async connection =>
+                    {
+                        await connection.SendResponseAsync($"HTTP/1.1 200 OK\r\nDate: {DateTimeOffset.UtcNow:R}\r\nContent-Length: 16000\r\n\r\n");
```

Comment:
You're postponing the timeout to the response body portion of HTTP exchange, right? If so, `HttpClient.GetAsync` will succeed as it by default reads only response headers and doesn't wait for the body.

Reply:
> `HttpClient.GetAsync` will succeed as it by default reads only response headers and doesn't wait for the body

`HttpClient.GetAsync` is passing `HttpCompletionOption.ResponseContentRead` to `HttpClient.SendAsync` which should enforce reading and buffering the response body. Or am I missing something?